service/src/main/java/uk/gov/mca/beacons/api/beacon/application/BeaconService.java

```diff
@@ -124,7 +119,33 @@ private ModelPatcher<Beacon> getPatcher() {
       .withMapping(Beacon::getModel, Beacon::setModel);
   }
 
-  public List<Beacon> findAll() {
-    return beaconRepository.findAll();
+  public Map<String, Integer> findHexIdsWithDuplicates(
+    int batchSize,
+    int numberAlreadyTaken
+  ) {
+    List<Beacon> batchOfBeacons = getBatch(batchSize, numberAlreadyTaken);
```

Comment:
Curious we're getting all the beacons here, do we not only care about what the duplicate factor is (hexId, and how many of them there are?)